### PR TITLE
[WIP] fix building on ubuntu 22.04

### PR DESCRIPTION
**Pull request by PiRK**
There is a missing header in a source file and some dependencies are not mentioned in the documentation.

### CODE REVIEW

**Comment by PiRK** [comment]
Something is still missing for flatbuffers. If I install it using the ubuntu packages, I'm getting compilation errors (probably because v1.11 is too old). If I install it from sources, I'm getting the following cmake error:
```
CMake Error at src/nng_interface/CMakeLists.txt:15 (find_package):
  Could not find a package configuration file provided by "Flatbuffers" with
  any of the following names:

    FlatbuffersConfig.cmake
    flatbuffers-config.cmake

  Add the installation prefix of "Flatbuffers" to CMAKE_PREFIX_PATH or set
  "Flatbuffers_DIR" to a directory containing one of the above files.  If
  "Flatbuffers" provides a separate development package or SDK, be sure it
  has been installed.


```

**Review by shammahatdivvy — APPROVED** [review]

**Comment by shammahatdivvy** [comment]
This is likely what you need: https://github.com/LogosFoundation/lotusd/pull/243/files

**Comment by shammahatdivvy** [comment]
@PiRK Tobias says you tried version 2 of flatbuffers already?

**Comment by PiRK** [comment]
> @PiRK Tobias says you tried version 2 of flatbuffers already?

Let me try again. Maybe i built from the master.

**Comment by PiRK** [comment]
With 2.0.0 it works.

I updated the build instructions to get the release source package instead of cloning the repository.

**Review by schancel — APPROVED** [review]
